src/instruments/internal/instrumentsreader.cpp

```diff
@@ -242,7 +243,7 @@ InstrumentTemplate InstrumentsReader::readInstrumentTemplate(Ms::XmlReader& read
         } else if (reader.name() == "transposeDiatonic") {
             instrument.transpose.diatonic = reader.readElementText().toInt();
         } else if (reader.name() == "instrumentId") {
-            instrument.id = reader.readElementText();
+            instrument.musicXMLid = reader.readElementText();
```

Comment:
Why are both `"instrumentId"` and `"musicXMLid"` going to `instrument.musicXMLid`?

Reply:
"`instrumentId`" is the unique identification, that's the one which we really need. "`muscXMLid`" is still used somewhere, e.g. when reading a score which doesn't have a `id` specified for an instrument. Then `musicXMLid` is used for guess what instrument it should be. It is also used for reading MU1 and MU2 scores.
So it is best keep this parameter (and up-to-date).

Reply:
Shouldn't `"instrumentId"` then go to `instrument.id` instead of `instrument.musicXMLid`?

Reply:
No, in the `Instrument` section of the score file, the MusicXMLid is called `instrumentId` (because was used as an id in older version of MuseScore. The same MusicXMLid is defined in `instruments.xml` with the `musicXMLid` tag.
An example:
In `instruments.xml` the Flute is defined as
```
<Instrument id="flute">
    <family>flutes</family>
    <longName>Flute</longName>
    <shortName>Fl.</shortName>
    <description>Standard Concert Flute</description>
    <musicXMLid>wind.flutes.flute</musicXMLid>
           .
           .
```
In the score file the same Flute appears as 
```
   <Instrument id="flute">
    <longName>Flute</longName>
    <shortName>Fl.</shortName>
           .
           .
    <instrumentId>wind.flutes.flute</instrumentId>
           .
           .
```
But welcome in the wonder world of instrument id's :wink: Here is some history....

Reply:
Ah, now I understand! Thanks!

Reply:
My pleasure :smiley: 

This drove me crazy last year, all these id's and in the meantime changing names.

Reply:
For MU4, I think we would be justified in changing `<instrumentId>` to `<MusicXMLid>` in the MSCX file format and using compatibility code to deal with old scores during import. There will already be other changes in the file format (e.g. for the Smart Tempos project), so it might be worth tidying up the syntax here and potentially [a few other places](https://musescore.org/en/node/296654) too.

Reply:
Changing the tag `<instrumentId>` into '<MusicXMLid>` would be nice and makes thing more clear :smile: